rmw/include/rmw/types.h

```diff
@@ -73,6 +91,13 @@ typedef struct RMW_PUBLIC_TYPE rmw_client_t
   const char * service_name;
 } rmw_client_t;
 
+typedef struct RMW_PUBLIC_TYPE rmw_event_t
+{
+  const char * implementation_identifier;
+  void * data;
```

Comment:
What is in this `data`? Looking at the review for rcl makes me think it's not just information about an event but all sorts of additional metadata. Is this just some generic pointer to a bunch of data used by the rmw_impl?

Reply:
One of the examples I see for data is the ConnextCustomEventInfo in rmw_connext. My guess is this has to be void * due to the different DDS implementations? It might be worth having a discussion to abstract this into a concrete type rather than void *.

Reply:
So is it a pointer to reference data for the rmw implementation to use or is it a pointer to data for things above the rmw layer to use? No documentation, no data type, and a pretty generic name make it hard to understand.

Reply:
> So is it a pointer to reference data for the rmw implementation to use or is it a pointer to data for things above the rmw layer to use? No documentation, no data type, and a pretty generic name make it hard to understand.

I don't disagree, as the above seems to be the theme for this feature. Right now I'm confusing this with the void * event found in rmw_connext (I don't think we need void * there), so need to figure out what should be done here.

Reply:
It is a pointer to data for the rmw implementation to use, not for things above the `rmw` layer to use.